module/item/item-sheet.js

```diff
@@ -280,6 +283,45 @@ Hooks.on("renderChatMessage", (message, html, data) => {
     }, 500);
 });
 
+// trading in food buff Items
+Hooks.on("renderChatMessage", (message, html, data) => {
+    setTimeout(() => {
+        $(html).find(".use-food-buff").on("click", (event) => consumeBuff(event));
+    }, 500);
+});
+
+export async function consumeBuff(event){
+	//prevent the default action of the button
+	event.preventDefault();
+
+	// Get the item ID and name from the button element's data-item-id and data-item-name attributes
+	const itemId = event.currentTarget.dataset.item;
+	const itemName = event.currentTarget.dataset.itemName;
+	const parentId = event.currentTarget.dataset.itemParentid;
+
+	//disable the button
+	event.currentTarget.disabled = true;
+
+	//find the actor with id parentId
+	const actor = game.actors.get(parentId);
+	console.log(actor);
+
+	console.log(`${actor.name} consumed the food buff - ${itemName}!`);
+
+	const buffArray = actor.system.digestionBuff.split(", ");
+
+	//remove the buff from the array
+	const index = buffArray.indexOf(itemName);
+	if (index > -1) {
+		buffArray.splice(index, 1);
+	}
+
+	//join the Array back to a string
+	const newBuffString = buffArray.join(", ");
```

Comment:
```suggestion
	const buffArray = actor.system.digestionBuff.split(", ");

        // Filter buff out of Array
        const newBuffArray = buggArray.filter(b => b.trim().toLowerCase() != itemName.trim().toLowerCase())

	// Join the Array back to a string
	const newBuffString = newBuffArray .join(", ");
```
Is one option, you could also do it all inline

`const newString = actor.system.digestionBuff.split(", ").filter(b => b.trim().toLowerCase() != itemName.trim().toLowerCase()).join(', ');`